pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -228,6 +228,23 @@ stdenv.mkDerivation ({
         configureFlags+=" --extra-lib-dirs=$p/lib"
       fi
     done
+  '' + (optionalString stdenv.isDarwin ''
+    # Work around a limit in the Mac OS X Sierra linker on the number of paths
+    # referenced by any one dynamic library:
+    #
+    # Create a local directory with symlinks of the *.dylib (Mac OS X shared
+    # libraries) from all the dependencies.
+    local dynamicLinksDir="$out/lib/links"
+    mkdir -p $dynamicLinksDir
+    local foundDylib=false
```

Comment:
Sorry about that; sent out #25602 to remove the line.

Originally I was going to add more complicated logic to handle the different Linux/Mac cases (since the former uses a different extension for shared libraries).  But the call to `stdenv.isDarwin` has an equivalent effect.